File: App/Backend/src/models/module.py
from sqlalchemy import Boolean, ForeignKey, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from Backend.database import Base


class Module(Base):
    __tablename__ = "modules"

    module_id: Mapped[int] = mapped_column(
        Integer,
        primary_key=True,
        autoincrement=True
    )

    course_id: Mapped[int] = mapped_column(
        Integer,
        ForeignKey("courses.course_id"),
        nullable=False
    )

    is_published: Mapped[bool] = mapped_column(
        Boolean,
        default=False
    )

    published_by: Mapped[str | None] = mapped_column(
        String(10),
        ForeignKey("users.uid"),
        nullable=True
    )

    module_name: Mapped[str] = mapped_column(
        String(150),
        nullable=False
    )

    description: Mapped[str | None] = mapped_column(
        Text,
        nullable=True
    )

File: App(SpringBoot)/src/main/java/com/Project/LearningManagementSystem/entity/Module.java
package com.Project.LearningManagementSystem.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "modules")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor
public class Module {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "module_id")
    private Integer moduleId;

    @Column(name = "course_id", nullable = false)
    private Integer courseId;

    @Column(name = "is_published", nullable = false)
    private boolean published = false;

    @Column(name = "published_by", length = 10)
    private String publishedBy;

    @Column(name = "module_name", length = 150, nullable = false)
    private String moduleName;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;
}
